### DISCUSSION TITLE
Add environment variable to existing matrix (TravisCI) migration

### DISCUSSION THREAD

**Discussion by buhtz** [woodpecker-ci/woodpecker#3317] [Ask a Question] [answered]
I don't have WoodpeckerCI at hand but plan to migrate to Codeberg.org which do host an instance. The config seems quit similar to TravisCI. I burn a lot of energy and hours at TravisCI.

So I would like to point out one of my conceptual problems with Travis and want to find out if Woodpecker handle this differently.

There is a quiet simple config for Python.

```
language: python
os: linux
dist: focal

arch:
 - amd64
 - ppc64le

env:
  PYTHONUNBUFFERED=1

python:
  - "3.8"
  - "3.9"
  - "3.11"
```

This will result in 6 jobs (3 Python versions x 2 architectures) all using PYTHONUNBUFFERED.

Now I want to add another environment variable to two of the jobs running Python 3.8.

```
jobs:
  include:
    - python: "3.8"
      env: FOO=bar
```

My problem on TravisCI and its logic is that this doesn't add this variable to the existing job but it simply creates a 7th job not using PYTHONUNBUFFERED (which is IMHO configured on a global level) but FOO.

How would I solve this with Woodpecker?

One important thing here is to keep the matrix implicit. I just give a list of 3 Python versions and 2 architectures. The matrix (6 jobs) is created implicit and automatic. I don't want to change that. Most TravisCI example I could find to start naming each of the 6 jobs explicit via "include:". So you will have 6 "include" repeating some of the same configs. When you want to modify something on a global level (e.g. modify the value of PYTHONUNBUFFERED) you have to edit all 6 includes. This redundancy is IMHO unneeded and gives a high risk of introducing undiscovered bugs.

**Comment by qwerty287** [comment, marked as answer]
You could use something like this:

```yaml
matrix:
    include:
        - PYTHON: "3.8"
          FOO: bar
        - PYTHON: "3.9"
        - PYTHON: "3.10"
```

**Reply by buhtz** [reply]
Thanks for the answer. Then this is the same uncomfortable approach I have seen on Travis.
In your solution you do explicit name 3 cells of the matrix just to add an environment variable into one of them. It is not a matrix anymore. It is just a list of 3 jobs.

**Reply by qwerty287** [reply]
That's not possible yet. Feel free to open a new feature request for it
